### PR TITLE
Donator commands

### LINKED ISSUE #733: Donator perk ideas
What about a custom death/respawn message?

### PR DESCRIPTION
**Pull request by mheguy**
~~Requires https://github.com/Refactorio/RedMew/pull/713~~
~~Requires https://github.com/Refactorio/RedMew/pull/757~~
The following maps will need to shut down before I can transition to tables:
- [x] Redmew - Valentine's Day
- [x] Redmew - Crash Site
- [x] Redmew - Diggy - Redux
- [x] RedMew - Line and Trees
- [x] RedMew - Tetris

Closes #733

### CODE REVIEW

File: features/rank_system.lua [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+        return
+    end
+    local player_name = player.name
+
+    if Public.equal(player_name, Ranks.guest) then
+        guests[player_name] = true
+    end
+
+    --- Fix for legacy name storage
+    local lowerCaseName = player_name:lower()
```
Comment (linaori) [inline features/rank_system.lua]: 🙈 🙉 🙊

**Review by linaori — DISMISSED** [review]
Pretty big change, might have missed a few things but looks good from what I can tell

**Comment by mheguy** [comment]
This just looks big because it's dependent on #713 and I have it rebased in.

**Comment by mheguy** [comment]
I should expand this to allow adding messages (`/donator-message add <message>`) to their list as well as the ability to view a list of their messages (`/donator-message list`) and remove messages they no longer want (`/donator-message delete <number>`).

**Comment by mheguy** [comment]
I'm content with where this is at now. It does require https://github.com/Refactorio/RedMew/pull/757 now though, to allow a transition from strings to tables.

**Comment by mheguy** [comment]
Before merging, this requires the extant messages data to be changed from strings to tables.
(More a note for myself so I don't forget)

**Comment by mheguy** [comment]
Note to me: add #733 into this

**Review by grilledham — APPROVED** [review]

**Pull request merged**